### PR TITLE
feat: implement ConsoleCommandListener

### PR DESCRIPTION
**Pull request by pforesi**
This pull request refactors the handling of OS signals for daemon commands by introducing a dedicated `ConsoleCommandListener` that responds to Symfony's `ConsoleSignalEvent`. The signal handling logic is moved out of `DaemonCommand`, and comprehensive unit and integration tests are added to ensure correct behavior. The documentation is also updated to describe the new testing strategy and best practices.

**Event-driven signal handling and listener introduction:**

* Added a new `ConsoleCommandListener` class that listens for `ConsoleSignalEvent` and triggers shutdown on `DaemonCommand` instances when receiving SIGTERM or SIGINT, using Symfony's event system and PHP 8 attributes. (`src/EventListener/ConsoleCommandListener.php`)
* Removed the direct `handleSignal` method and related signal handler registration from `DaemonCommand`, making signal handling fully event-driven. (`src/Command/DaemonCommand.php`) [[1]](diffhunk://#diff-b06933e6c6399ab568c43e4cd46376aa7040e1a7e093470693d1066be564147bL156-L159) [[2]](diffhunk://#diff-b06933e6c6399ab568c43e4cd46376aa7040e1a7e093470693d1066be564147bL422-L435)

**Testing improvements:**

* Added comprehensive unit tests for the new listener, covering all signal and command-type combinations, attribute configuration, and edge cases. (`tests/Unit/EventListener/ConsoleCommandListenerTest.php`)
* Added integration tests to verify the listener's behavior within Symfony's event system, including handling multiple listeners and signals. (`tests/Integration/EventListener/ConsoleCommandListenerIntegrationTest.php`)
* Updated the test fixture `DaemonCommandConcrete` to include an `isShutdownRequested()` helper for easier assertions in tests. (`tests/Fixtures/Command/DaemonCommandConcrete.php`)
* Removed obsolete signal handling tests from the old `DaemonCommandTest`. (`tests/Unit/Command/DaemonCommandTest.php`)

**Documentation updates:**

* Added a detailed markdown document describing the new listener's tests, structure, best practices, and coverage statistics. (`docs/CONSOLE_COMMAND_LISTENER_TESTS.md`)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,14 +4,11 @@ on: [push, pull_request]
...
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        php-version: ['8.1', '8.2']
-        symfony-version: ['^6.1', '^7.0']
-        exclude:
-          - php-version: '8.1'
-            symfony-version: '^7.0'
+        php-version: ['8.2', '8.3', '8.4']
+        symfony-version: ['^6.4', '^7.0']
```
Comment (Oliboy50) [inline .github/workflows/ci.yml:11]: I don't really understand the purpose of this PR, because [you talked about Symfony 7.3](https://github.com/BedrockStreaming/DaemonBundle/issues/43), but this bundle seemed to already be tested against Symfony `^7.0`, so:

did our tests were not covering enough things and were not proving that Symfony 7.0 was not supported correctly? 😢 
or did Symfony introduced a breaking change between `7.0` and `7.4`? 😢

**Review by Oliboy50 — APPROVED** [review]
🤞 👏 🤷

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pforesi) [reply .github/workflows/ci.yml:11]: As explain in the opened issue, the bug appears from symfony 7.3:
When updating to sf >= 7.3 projects based on DaemonBundle fails since symfony [Command](https://github.com/symfony/symfony/blob/bb2e6f9cb6e7942b7bb983a519f040e870bc73d9/src/Symfony/Component/Console/Command/Command.php#L35) implements the [SignalableCommandInterface](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/Console/Command/SignalableCommandInterface.php) from its 7.3 version.
This interface define the function:
`public function handleSignal(int $signal, int|false $previousExitCode = 0): int|false;`

that conflict with the one defined into [DaemonCommand](https://github.com/BedrockStreaming/DaemonBundle/blob/90b03db67cc56c2d11c8c380ad7076f08b292003/src/Command/DaemonCommand.php#L425)

To resolve this issue I propose to handle signal by adding a src/EventListener/ConsoleCommandListener.php that hanlde the signal instead of to add signal handler directly in the DaemonCommand contructor:
`        // Add the signal handler
        pcntl_signal(SIGTERM, [$this, 'handleSignal']);
        pcntl_signal(SIGINT, [$this, 'handleSignal']);
`

Concerning the ci.yml I just updated the matrix by adding php 8.3, php 8.4 and removing 8.1 as long as [It will be no more supported at the end of the month of december](https://www.php.net/supported-versions.php)

**Comment by pforesi** [comment]
Just squashed my commits

**Comment by pforesi** [comment]
UP, please may the 2nd reviewer review, accept or reject. Our team need it, elsewhere we will go on forked solution.
Thanks in advance :)

**Review by t-geindre — APPROVED** [review]

**Comment by Oliboy50** [comment]
@pforesi thank you for your time and efforts here

this has been released in [v9.0.0](https://packagist.org/packages/m6web/daemon-bundle#v9.0.0) (this was considered as a breaking change since we're dropping a public function from this bundle)
